patches/server/0840-Fix-mobs-spawning-outside-hard-despawn-range.patch

```diff
@@ -0,0 +1,22 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Gaider10 <>
+Date: Mon, 30 Aug 2021 16:10:51 +0300
+Subject: [PATCH] Fix mobs spawning outside hard despawn range
+
+
+diff --git a/src/main/java/net/minecraft/world/level/NaturalSpawner.java b/src/main/java/net/minecraft/world/level/NaturalSpawner.java
+index bacd75f67f783f49208a74501cc7e6e7485010a4..603df526db7246283fff24dd2799ebd94ba25b7d 100644
+--- a/src/main/java/net/minecraft/world/level/NaturalSpawner.java
++++ b/src/main/java/net/minecraft/world/level/NaturalSpawner.java
+@@ -401,7 +401,10 @@ public final class NaturalSpawner {
+     }
+ 
+     private static boolean isValidPositionForMob(ServerLevel world, Mob entity, double squaredDistance) {
+-        return squaredDistance > (double) (entity.getType().getCategory().getDespawnDistance() * entity.getType().getCategory().getDespawnDistance()) && entity.removeWhenFarAway(squaredDistance) ? false : entity.checkSpawnRules((LevelAccessor) world, MobSpawnType.NATURAL) && entity.checkSpawnObstruction((LevelReader) world);
++        // Paper start - fix mobs spawning outside hard despawn range
++        int despawnDistance = entity.isPersistenceRequired() || entity.requiresCustomPersistence() ? entity.getType().getCategory().getDespawnDistance() * entity.getType().getCategory().getDespawnDistance() : world.paperConfig.hardDespawnDistance;
```

Comment:
This needs to be updated. despawn distances were just made per-category, so this won't compile.

Reply:
Fixed